PWGHF/HFC/TableProducer/correlatorDsHadrons.cxx

```diff
@@ -0,0 +1,341 @@
+
+// histogram axes definition
+AxisSpec axisMassD = {300, 1.7, 2.2, ""};
+AxisSpec axisPhi = {128, -o2::constants::math::PIHalf, 3. * o2::constants::math::PIHalf, ""};
+AxisSpec axisEta = {100, -2., 2., ""};
+AxisSpec axisY = {100, -2., 2., ""};
+AxisSpec axisPtD = {180, 0., 36., ""};
+
+using MCParticlesPlus3Prong = soa::Join<aod::McParticles, aod::HfCand3ProngMcGen>;
+
+/// Ds-Hadron correlation pair builder - for real data and data-like analysis (i.e. reco-level w/o matching request via MC truth)
+struct HfCorrelatorDsHadrons {
+  Produces<aod::DsHadronPair> entryDsHadronPair;
+  Produces<aod::DsHadronRecoInfo> entryDsHadronRecoInfo;
+
+  Configurable<int> selectionFlagDs{"selectionFlagDs", 7, "Selection Flag for Ds"};
+  Configurable<int> applyEfficiency{"applyEfficiency", 1, "Flag for applying D-meson efficiency weights"};
+  Configurable<double> yCandMax{"yCandMax", 0.8, "max. cand. rapidity"};
+  Configurable<double> etaTrackMax{"etaTrackMax", 0.8, "max. eta of tracks"};
+  Configurable<double> dcaXYTrackMax{"dcaXYTrackMax", 1., "max. DCA_xy of tracks"};
+  Configurable<double> dcaZTrackMax{"dcaZTrackMax", 1., "max. DCA_z of tracks"};
+  Configurable<double> ptCandMin{"ptCandMin", 1., "min. cand. pT"};
+  Configurable<double> ptTrackMin{"ptTrackMin", 0.3, "min. track pT"};
+  Configurable<double> ptTrackMax{"ptTrackMax", 50., "max. track pT"};
+  Configurable<double> multMin{"multMin", 0., "minimum multiplicity accepted"};
+  Configurable<double> multMax{"multMax", 10000., "maximum multiplicity accepted"};
+  Configurable<std::vector<double>> binsPt{"binsPt", std::vector<double>{o2::analysis::hf_cuts_ds_to_k_k_pi::vecBinsPt}, "pT bin limits for candidate mass plots and efficiency"};
+  Configurable<std::vector<double>> efficiencyD{"efficiencyD", std::vector<double>{vecEfficiencyDmeson}, "Efficiency values for Ds meson"};
+
+  Partition<soa::Join<aod::HfCand3Prong, aod::HfSelDsToKKPi>> selectedDsCandidates = aod::hf_sel_candidate_ds::isSelDsToKKPi >= selectionFlagDs || aod::hf_sel_candidate_ds::isSelDsToPiKK >= selectionFlagDs;
+  Partition<soa::Join<aod::HfCand3Prong, aod::HfSelDsToKKPi, aod::HfCand3ProngMcRec>> recoFlagDsCandidates = aod::hf_sel_candidate_ds::isSelDsToKKPi >= selectionFlagDs || aod::hf_sel_candidate_ds::isSelDsToPiKK >= selectionFlagDs;
+
+  HistogramRegistry registry{
+    "registry",
+    {{"hPtCand", "Ds,Hadron candidates;candidate #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {axisPtD}}},
+     {"hPtProng0", "Ds,Hadron candidates;prong 0 #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {axisPtD}}},
+     {"hPtProng1", "Ds,Hadron candidates;prong 1 #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {axisPtD}}},
+     {"hPtProng2", "Ds,Hadron candidates;prong 2 #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {axisPtD}}},
+     {"hSelectionStatus", "Ds,Hadron candidates;selection status;entries", {HistType::kTH1F, {{11, -0.5, 10.5}}}},
```

Comment:
Why do you need 11 bins for the status?

Reply:
I've set 11 bins just to control the flag of selection status for Ds which is 7

Reply:
So why do you not have 7 bins from 0 to 7 then?